### PR TITLE
initial template

### PR DESCRIPTION
**Pull request by fishnix**

### CODE REVIEW

**Comment by damnski** [comment]
hawt

File: README.md.tmpl [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+```
+GET {{ .UrlBase }}/ping
+GET {{ .UrlBase }}/version
+GET {{ .UrlBase }}/metrics
+```
+
+## Authentication
+
+Authentication is accomplished via a pre-shared key.  This is done via the `X-Auth-Token` header.
```
Comment (tenyo) [inline README.md.tmpl:15]: 👀 Authentication is accomplished via an encrypted pre-shared key in the `X-Auth-Token` header

File: api/handlers_test.go.tmpl [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+package api
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/YaleSpinup/ec2-api/common"
```
Comment (tenyo) [inline api/handlers_test.go.tmpl:24]: 👀

File: api/handlers_test.go.tmpl [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+)
+
+func TestPingHandler(t *testing.T) {
+	req, err := http.NewRequest("GET", "/v1/ec2/ping", nil)
```
Comment (tenyo) [inline api/handlers_test.go.tmpl:28]: 👀

File: api/handlers_test.go.tmpl [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+	expected := `pong`
+	if rr.Body.String() != expected {
+		t.Errorf("handler returned unexpected body: got %v want %v",
+			rr.Body.String(), expected)
+	}
+}
+
+func TestVersionHandler(t *testing.T) {
+	req, err := http.NewRequest("GET", "/v1/ec2/version", nil)
```
Comment (tenyo) [inline api/handlers_test.go.tmpl:51]: 👀

File: api/server.go.tmpl [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+	// setup server context with cancellation
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	if config.Org == "" {
+		return errors.New("'org' cannot be empty in the configuration")
+	}
+
+	s := server{
+		// ec2Services:         make(map[string]ec2.EC2),
```
Comment (tenyo) [inline api/server.go.tmpl:56]: Is this left as an example?

File: k8s/README.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+* base64 encode the configuration `cat config/config.json | base64 -w0`
+
+* copy output of `config.json` secret into `k8s-config.yaml`
+
+## develop
+
+* run `skaffold dev` in the root of the project
+
+* update your `hosts` file to point spindev.internal.yale.yale.edu to localhost
```
Comment (tenyo) [inline k8s/README.md:36]: 👀 yale yale

File: api/server.go.tmpl (hunk 5 above)
Comment (fishnix) [reply api/server.go.tmpl:56]: yeah exactly - i figured i would leave it there

**Pull request merged**